configs/myconfigs/sample_2d.py

```diff
@@ -0,0 +1,158 @@
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Authors: Jason Lowe-Power, Brian Coutinho
+
+""" This script is an example of multi-threaded statistical sampling
+    where we simulate a sample multiple times to cover several possible
+    thread interleavings. Random pertubations are added during the
+    simulation to exercise mutliple execution paths/interleavings.
+
+    The sampling process thus works as two nested for-loops,
+    the outer one for each sample point in the program and
+    the inner one for mulitple observations at each sample point.
+
+    This script takes three arguments, the total number of instructions in the
+    ROI (which you can find by using runkvm.py), the number of sample
+    points you want to take during the execution, and the number of runs
+    to make at each sample point.
+
+    The output is a set of two levels of directories in the outdir.
+    At the top level is directory for each sample point which in turn
+    contains multiple sub-directories for each random observation.
+
+"""
+
+import os
+import sys
+import time
+
+import m5
+from m5.objects import *
+from m5.ticks import fromSeconds
+from m5.util.convert import toLatency
+
+sys.path.append('configs/common/') # For the next line...
+import SimpleOpts
+
+from system import MySystem
+
+# Sampling library
+from sampling import sampleROI, checkSystem
+
+SimpleOpts.add_option("--script", default='',
+                      help="Script to execute in the simulated system")
+SimpleOpts.set_usage("usage: %prog [options] roi_instructions \
+samples=1 runs=1")
+
+
+
+if __name__ == "__m5_main__":
+    (opts, args) = SimpleOpts.parse_args()
+
+    # create the system we are going to simulate
+    system = MySystem(opts)
+
+    # Check if the system is compatible with sampling library
+    checkSystem(system)
```

Comment:
Probably should do this before you assume that the System constructor takes one parameter which is "opts". Maybe do this in two phases?

Or, you could wrap the these two lines in a try/catch and raise an exception. That's probably the best way to do this.